java/code/src/com/redhat/rhn/frontend/action/common/DownloadFile.java

```diff
@@ -282,7 +282,8 @@ else if (ksession != null) {
 
         if (map.containsKey(CHILD) && !Config.get().getBoolean("ks_restrict_child_channels")) {
             Channel child = ChannelFactory.lookupByLabel(map.get(CHILD));
-            if (child == null || tree == null || !child.getParentChannel().equals(tree.getChannel())) {
+            if (child == null || tree == null || child.getParentChannel() == null ||
+                    !child.getParentChannel().equals(tree.getChannel())) {
```

Comment:
sorry but how can a child channel have not parent? If it doesn't have a parent channel then it's not a child :D

Reply:
looks like some reeeally strange  (and not supported...and pro)  setup might allow it. basically autoyast (that is supposed to be used just for child channels), was used with a parent channel: so  the child here was actually a parent. 

Besides the hacky way to reach this point, the code should prevents any NPE

Reply:
>the code should prevents any NPE

Sure, I was just surprise to see a child channel without a parent :)